File: stack/evalRPN.py
class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        
        stack = []

        for item in tokens:
            if item=="+":
                stack.append(stack.pop()+stack.pop())
            elif item=="-":
                a, b=stack.pop(),stack.pop()
                stack.append(b-a)

            elif item=="*":
                stack.append(stack.pop()*stack.pop())

            elif item=="/":
                a, b=stack.pop(),stack.pop()
                stack.append(b/a)
            else:
                stack.append(int(item)) 
        print(stack)
Solution().evalRPN(["2","1","+","3","*"])

File: stack/evalRPN.go
package main

import (
	"fmt"
	"strconv"
)

func evalRPN(tokens []string) int {
	stack := []int{}
	ops := map[string]func(a, b int) int{
		"+": func(a, b int) int { return a + b },
		"-": func(a, b int) int { return b - a },
		"*": func(a, b int) int { return a * b },
		"/": func(a, b int) int { return b / a },
	}
	for _, token := range tokens {
		op, ok := ops[token]
		if ok {
			a, b := stack[len(stack)-1], stack[len(stack)-2]
			res := op(a, b)
			fmt.Println(a, b, res, token, 13/5)
			stack = append(stack[:len(stack)-2], res)
			continue
		}
		n, _ := strconv.Atoi(token)
		stack = append(stack, n)

	}
	fmt.Println(stack)
	return 1
}
func main() {

	evalRPN([]string{"4", "13", "5", "/", "+"})

}
